### DISCUSSION TITLE
Mapbox marker with ID 'studentMarker' was initialized multiple times. This can cause unexpected behaviour

### DISCUSSION THREAD

**Discussion by EntrenaEnCasa** [AlexLavoie42/Nuxt-Mapbox#102] [Q&A]
I've encountered a recurring error where a Mapbox marker gets initialized multiple times, or at least, that's what the error says, which is leading to unexpected behaviour in my app. The issue arises under a specific set of circumstances that involve user authentication flow—specifically, logging out and then logging back into the application, followed by navigating to the component that incorporates the Mapbox component. Curiously, refreshing the page resolves the issue, which suggests that it might be tied to the state management or component lifecycle within my app.

However, this problem seems to be exclusive to the Mapbox component; I haven't observed similar behaviour with any other components in my application. Given the peculiarity of the reproduction steps and the context in which the issue occurs, I'm uncertain if this is related to my implementation of the login/logout functionality or if it's an issue with the Mapbox component integration in a Nuxt environment.

Have you experienced a similar issue, or do you have insights into potential causes? Any suggestions on how to debug or fix this would be greatly appreciated.

here's the issue in the console:

![image](https://github.com/AlexLavoie42/Nuxt-Mapbox/assets/137846923/b0234728-c435-4da2-b8f5-de88f1b884d0)

**Comment by AlexLavoie42** [comment]
It's hard to say without being able to look closer into your application unfortunately. I have seen issues like this arrive when using the same map or marker ID on multiple pages so be sure to double check that is not the case. You could also try [disabling persistent mode](https://alexlavoie42.github.io/Nuxt-Mapbox/configuration/mapbox-config#persistent-map-instances) as a simple fix.

This could also very well be a bug in the module. If you are able to reproduce this in a new project I would be able to investigate further.
